fieldpath/set.go

```diff
@@ -503,3 +652,50 @@ func (s *SetNodeMap) Leaves() *SetNodeMap {
 	}
 	return out
 }
+
+// Filter defines an interface for excluding field paths from a set.
+// NewExcludeFilter can be used to create a filter that removes
+// excluded field paths.
+// NewPatternFilter can be used to create a filter that removes all fields except
+// the fields that match a field path pattern. PrefixPattern and MustPrefixPattern
+// can be used to define field path patterns.
+type Filter interface {
+	// Filter returns a filtered copy of the set.
+	Filter(*Set) *Set
+}
+
+// NewExcludeFilter returns a filter that removes field paths in the exclude set.
+func NewExcludeFilter(exclude *Set) Filter {
```

Comment:
Maybe `NewExcludeSetFilter` so the names fit a pattern of `New<In/Ex>clude<Source>Filter()` ?